realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -917,11 +917,33 @@ public void testWriteCopyTo() throws IOException {
         }
     }
 
+
+    public void testCompactRealmFileThrowsIfOpen() throws IOException {
+        try {
+            Realm.compactRealmFile(getContext());
+        } catch (IllegalStateException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testCompactEncryptedRealmFile() {
+        String REALM_NAME = "enc.realm";
+        Realm.deleteRealmFile(getContext(), REALM_NAME);
+        byte[] key = new byte[64];
+        new Random(42).nextBytes(key);
+        Realm realm = Realm.getInstance(getContext(), REALM_NAME, key);
+        realm.close();
+        assertTrue(Realm.compactRealmFile(getContext(), REALM_NAME, key));
+    }
+
     public void testCompactRealmFile() throws IOException {
+        testRealm.close();
```

Comment:
Not a good idea. This can potentially screw up the reference counting. We should probably use a completely other Realm file.